src/drivers/intel/hda/hda-dma.c

```diff
@@ -407,8 +407,7 @@ static int hda_dma_enable_unlock(struct dma_chan_data *channel)
 	}
 
 	/* start link output transfer now */
-	if (channel->direction == DMA_DIR_MEM_TO_DEV &&
-	    !(hda_chan->state & HDA_STATE_RELEASE))
+	if (channel->direction == DMA_DIR_MEM_TO_DEV)
```

Comment:
I wasn't able to follow that simplification.

Reply:
the only thing release ever did was to start host DMA but we never pause/release the host DMA in the first place, so the preovious patch removed that. With that, release literally does nothing now

Reply:
but we still set the HDA_STATE_RELEASE bit? I am officially confused!

Reply:
that bit is used to do 2 things: 1. releae host DMA which will never happen because we do no release host DMA. 2. prevent incrementing the firmware pointer for release which actually results in an xrun. But it isnt a register BIT just a state bit we used for the above 2 things. So its no longer needed

Reply:
so can we remove line 414 as well and even the definition of HDA_STATE_RELEASE?

Reply:
but I have removed the defitnion of HDA_STATE_RELEASE, no? not sure what 414 is, probably clearing the state and I' have removed that as well

Reply:
TBH, I would lump all the HDA DMA driver RELEASE removal into one patch.